Mathlib/Algebra/PresentedMonoid/Basic.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2024. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hannah Fechtner
+-/
+
+import Mathlib.Algebra.FreeMonoid.Basic
+import Mathlib.Algebra.Group.Submonoid.Operations
+import Mathlib.GroupTheory.Congruence.Hom
+
+/-!
+# Defining a monoid given by generators and relations
+
+Given a subset `rels` of relations of the free monoid on a type `α`, this file constructs the monoid
+given by generators `x : α` and relations `r ∈ rels`.
+
+## Main definitions
+
+* `PresentedMonoid rels`: the quot group of the free group on a type `α` by the steps-to closure
```

Comment:
Is "steps-to" closure a standard terminology? I've never heard of it.

Reply:
Oy, it is not! I went back to some textbooks, and it looks like the proper term is "one-step reduction"

I've updated the definitions (thanks to your comment, I realized I had other errors too!), though it feels clunky. I'm not sure how to best describe `rels`. It's not a set - can I just call it a relation? Or is it a function? Any advice would be much appreciated!

(I may also have better ideas in the morning)